[MAP] Box improvements

Incoming5643 opened the pull request:
Replaces the admittedly hacky "some closets are boxes on spawning" code in favor of a dedicated Closet/Box mapping object. This was actually the plan all along.

Manually places these box/closet objects over relevent maint closets on meta and box(station)

Also fixes #12006 the easter egg was occurring silently when <a href="http://i.ytimg.com/vi/AIkVVclDMoc/maxresdefault.jpg">ghosts and corpses</a> were the only thing that saw you exit the box.

Incoming5643 committed:
Replaces the admittedly hacky "some closets are boxes on spawning" code in favor of a dedicated Closet/Box mapping object. This was actually the plan all along.

Either this pull or a future pull will have map edits to replace most of the generic maint closets with these, but I don't want to do them until big remaps are still pending.

Incoming5643 commented:
Pulls dedicated to cardboard boxes: 3

duncathan commented on code/game/objects/effects/spawners/structure.dm:
```diff
@@ -31,3 +31,14 @@ again.
+	icon_state = "closet_or_box_spawner"
+
+/obj/effect/spawner/structure/closet_or_box/New()
+	if(prob(25))
```
you buffed the box probability unstated change :police: :police:

duncathan replied:
github emoji on mobile is hard ok

Jalleo replied:
Its :rotating_light: rotating_light use firefox for mobile(android) @duncathan to switch to desktop view easy with one refresh.

Incoming5643 replied:
I did so because the mapping object can be applied selectively instead of just applying to every generic closet. For example there's some generic closets in fitness that shouldn't be boxes ever. The average box per station coefficient (Avebox/station) roughly shouldn't change as a result.

:cop:

Incoming5643 committed:
fixes spaceman accidentally setting off the egg timer because a ghost/corpse saw them exit the box but no one else did.

Incoming5643 committed:
Meticulously designates all the proper maint closets for the box chance on box and Meta
Reduces the box chance back to 20% after doing a little testing on how many boxes are actually spawned on average at 25%

Incoming5643 commented:
>wait until after big map conflicts to do big manual box or closet replacements on the map
>forgot to update to master first

Fucked that one up pretty good!

Incoming5643 committed:
Merge branch 'master' of https://github.com/tgstation/-tg-station into BoxesOfLiberty

Conflicts:
	_maps/map_files/TgStation/tgstation.2.1.3.dmm

Incoming5643 committed:
Refixes closet|box spawners on boxstation

Incoming5643 commented:
Alright that should do it. Now that this is officially a map pull I hope it doesn't languish at bad as the initial box pull did.

Incoming5643 commented:
Absorbed into #12054
